Assets/_PlatformerDevelopment/Scripts/Managers/StateManager.cs

```diff
@@ -25,12 +25,23 @@ public interface IStateManager
         void SetState(State state);
     }
     
-    public class StateManager: IStateManager
+    public class StateManager : IStateManager 
```

Comment:
Had to make this a Singleton 
`public event Action<State> OnStateChanged;`

So that this would be shared, and also the interface does not allow static event unless we use C# 8.0, which this project doesn't seem to allow.